src/exo/LoopIR_scheduling.py

```diff
@@ -195,7 +195,41 @@ def map_e(self,e):
         return super().map_e(e)
 
 
+class _DoMergeReduce(LoopIR_Rewrite):
+    def __init__(self, proc, stmt1, stmt2):
+        if type(stmt1) is LoopIR.Assign and type(stmt2) is LoopIR.Reduce:
+            if stmt1.name != stmt2.name:
+                raise SchedulingError("expected the two statements to have the same lhs.")
+
+            if len(stmt1.idx) != len(stmt2.idx):
+                raise SchedulingError("expected the LHS indices to be the same.")
+            try:
+                for i, j in zip(stmt1.idx, stmt2.idx):
+                    Check_ExprEqvInContext(proc, [stmt1, stmt2], i, j)
+            except SchedulingError:
+                raise SchedulingError("expected the LHS indices to be the same.")
+
+            self.new_rhs = LoopIR.BinOp("+", stmt1.rhs, stmt2.rhs,T.i32, stmt1.srcinfo)
```

Comment:
We should also check data dependencies on the rhs

Reply:
And write tests for those cases, e.g.
```
a = b
a += a
```
does not rewrite to `a = a + b`, but rather `a = b + b`